indexing-service/src/main/java/org/apache/druid/indexing/common/task/AbstractTask.java

```diff
@@ -216,6 +232,23 @@ public void cleanUp(TaskToolbox toolbox, TaskStatus taskStatus) throws Exception
     }
   }
 
+  @Override
+  public boolean waitForCleanupToFinish()
+  {
+    try {
+      if (cleanupCompletionLatch != null) {
+        // block until the cleanup process completes
+        return cleanupCompletionLatch.await(30, TimeUnit.SECONDS);
```

Comment:
Is 30 seconds typically enough time for the cleanup to finish?
I think 5 minutes might be better as the cleanup seems to be doing a bunch of things - update status, update location, push task reports, .

It might also be good to
- log a warning message if the cleanup could not be finished in time and the latch returned `false`.
- log an info message for the time taken to finish the cleanup (or is it already being done in `cleanup` method?)

Reply:
I updated to 60 seconds, it won't take that long to finish cleanUp. warning message is also added.
the time taken to shutdown is recorded in `SingleTaskBackgroundRunner#stop`

Reply:
Yes, typically it would be fast, but in case it isn't, I believe it is better to wait a little longer since this change is going to affect all tasks. The cleanup seems to submit a couple of actions to the Overlord which could be slow in getting processed.

Reply:
updated to 5 mins